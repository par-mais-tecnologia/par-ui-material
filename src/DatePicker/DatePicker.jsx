import React, { PureComponent } from 'react'
import MuiPickersUtilsProvider from 'material-ui-pickers/utils/MuiPickersUtilsProvider'
import DatePickerMUI from 'material-ui-pickers/DatePicker'
import PropTypes from 'prop-types'
import FormHelperText from '@material-ui/core/FormHelperText/FormHelperText'

class DatePicker extends PureComponent {

  state = {}

  validationInstance = this.props.validator ? this.props.validator.validator.register(
    this
  ) : false

  handleOnBlur(event) {
    if (this.props.onBlur) {
      return this.props.onBlur(event)
    } else {
      return this.validationInstance ? this.validationInstance.validate(event) : false
    }
  }

  render () {
    const inputProps = {
      onBlur: this.handleOnBlur.bind(this)
    }
    const { dateUtilityLibrary } = this.props
    if (this.validationInstance) {
      this.state.errors = {
        ...this.validationInstance.error
      }
    } else {
      this.state.errors = { hasError: false }
    }
    const proxyProps = { ...this.props }
    delete proxyProps.dateUtilityLibrary

    return (
      <MuiPickersUtilsProvider utils={dateUtilityLibrary}>
        <DatePickerMUI
          {...proxyProps}
          {...inputProps}
          error={this.state.errors.hasError}
        />
        {this.state.errors.hasError ?
          <FormHelperText style={{color:'#f44336'}}>
            {this.state.errors.errorMessage}
          </FormHelperText> : ''}
      </MuiPickersUtilsProvider>
    )
  }
}

DatePicker.propTYpes = {
  utils: PropTypes.object
}

export default DatePicker
