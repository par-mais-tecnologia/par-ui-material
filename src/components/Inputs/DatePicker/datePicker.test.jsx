import React from "react";
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import DatePicker from './datePicker'
import MomentUtils from "material-ui-pickers/utils/moment-utils";
configure({ adapter: new Adapter() });

export const test = describe('DatePicker', () => {
        it('shallow DatePicker', () => {
            const wrapper = shallow(<DatePicker dateUtilityLibrary = {MomentUtils}/>);
            expect(wrapper).toBeTruthy();
        });

        it('shallow DatePicker match Snapshot', () => {
            const wrapper = shallow(<DatePicker dateUtilityLibrary = {MomentUtils}/>);
            expect(wrapper).toMatchSnapshot();
        });
    });