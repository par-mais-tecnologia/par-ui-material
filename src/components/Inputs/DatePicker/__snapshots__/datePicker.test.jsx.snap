// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker shallow DatePicker 2 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatePicker
    dateUtilityLibrary={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef
        dateUtilityLibrary={[Function]}
      />,
      "locale": undefined,
      "moment": undefined,
      "utils": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dateUtilityLibrary": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef
          dateUtilityLibrary={[Function]}
        />,
        "locale": undefined,
        "moment": undefined,
        "utils": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dateUtilityLibrary": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`DatePicker shallow DatePicker 3 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatePicker
    dateUtilityLibrary={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef
        dateUtilityLibrary={[Function]}
      />,
      "locale": undefined,
      "moment": undefined,
      "utils": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dateUtilityLibrary": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef
          dateUtilityLibrary={[Function]}
        />,
        "locale": undefined,
        "moment": undefined,
        "utils": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dateUtilityLibrary": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`DatePicker shallow DatePicker match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatePicker
    dateUtilityLibrary={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef
        dateUtilityLibrary={[Function]}
      />,
      "locale": undefined,
      "moment": undefined,
      "utils": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dateUtilityLibrary": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef
          dateUtilityLibrary={[Function]}
        />,
        "locale": undefined,
        "moment": undefined,
        "utils": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dateUtilityLibrary": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
